grails-app/services/com/recomdata/transmart/data/export/ExportMetadataService.groovy

```diff
@@ -258,47 +251,21 @@ class ExportMetadataService {
                     dataTypeName     : highDimRow.datatype.dataTypeDescription,
                     isHighDimensional: true,
-                    metadataExists   : true,
-
-                    subset1          : exporters.collect {
-                        [
-                                fileType         : "." + it.format,
-                                dataTypeHasCounts: true,
-                                dataFormat       : it.description,
-                                fileDataCount    : highDimRow.subset1 ? highDimRow.subset1.size() : 0,
-                                platforms        : platforms.subset1
-                        ]
-                    },
-                    subset2          : exporters.collect {
-                        [
-                                fileType         : "." + it.format,
-                                dataTypeHasCounts: true,
-                                dataFormat       : it.description,
-                                fileDataCount    : highDimRow.subset2 ? highDimRow.subset2.size() : 0,
-                                platforms        : platforms.subset2
-                        ]
-                    }
-            ]
-        }
-    }
-
-    /**
-     * Returns a list of unique platforms for a given set of subject sample mappings 
-     * @param assayList
-     * @return A list of unique platforms, each being a map with the keys
-     *              gplId
-     *              gplTitle
-     *              fileDataCount
-     */
-    private getPlatformsForSubjectSampleMappingList(Collection<Assay> assayList) {
-        if (!assayList)
-            return []
-
-        return assayList*.platform.unique().collect { platform ->
-            [
-                    gplId        : platform.id,
-                    gplTitle     : platform.title,
-                    fileDataCount: assayList.findAll { assay -> assay.platform.id == platform.id }.size()
+                    subset1          : [
+                            dataTypeHasCounts: true,
+                            exporters        : exporters.collect { [format: it.format, description: it.description] },
+                            patientsNumber   : highDimRow.subset1 ?
+                                    (highDimRow.subset1*.patientInTrialId).unique().size() : 0,
+                            ontologyTermKeys : highDimRow.subset1_hd_terms
+                    ],
+
+                    subset2          : [
+                            dataTypeHasCounts: true,
+                            exporters        : exporters.collect { [format: it.format, description: it.description] },
+                            patientsNumber   : highDimRow.subset2 ?
+                                    (highDimRow.subset2*.patientInTrialId).unique().size() : 0,
+                            ontologyTermKeys : highDimRow.subset2_hd_terms
+                    ]
             ]
         }
     }
```

Comment:
This class in general has a quite some duplicated code for subset1 and subset2 (which is even more worrying when you think some people are trying to implement arbitrary number of subsets). It would benefit from some generalization. This probably applies also to the maps communicated between the client and the server with json.